test/unit/ngsiv2/provisioning/removeProvisionedDevice-test.js

```diff
@@ -0,0 +1,179 @@
+    request = require('request'),
+    contextBrokerMock,
+    iotAgentConfig = {
+        logLevel: 'FATAL',
+        contextBroker: {
+            host: '192.168.1.1',
+            port: '1026',
+            ngsiVersion: 'v2'
+        },
+        server: {
+            port: 4041,
+            baseRoot: '/'
+        },
+        types: {},
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S'
+    };
+
+describe('Device provisioning API: Remove provisioned devices', function() {
+    var provisioning1Options = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+            method: 'POST',
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            },
+            json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionNewDevice.json')
+        },
+        provisioning2Options = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices',
+            method: 'POST',
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            },
+            json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/provisionAnotherDevice.json')
+        };
+
+    beforeEach(function(done) {
+        iotAgentLib.activate(iotAgentConfig, function() {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/registrations',
+                utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextAvailabilityRequests/registerProvisionedDevice.json'))
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
+                .reply(204);
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/registrations',
+                utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/contextAvailabilityRequests/registerProvisionedDevice2.json'))
+                .reply(201, null, {'Location': '/v2/registrations/6319a7f5254b05844116584d'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5